beater/handlers.go

```diff
@@ -105,11 +127,12 @@ func frontendHandler(pf ProcessorFactory, config *Config, report reporter) http.
 		ExcludeFromGrouping: regexp.MustCompile(config.Frontend.ExcludeFromGrouping),
 	}
 	return logHandler(
-		killSwitchHandler(config.Frontend.isEnabled(),
-			ipRateLimitHandler(config.Frontend.RateLimit,
-				corsHandler(config.Frontend.AllowOrigins,
-					processRequestHandler(pf, &prConfig, report,
-						decoder.DecodeUserData(decoder.DecodeLimitJSONData(config.MaxUnzippedSize), config.AugmentEnabled))))))
+		concurrencyLimitHandler(config,
+			killSwitchHandler(config.Frontend.isEnabled(),
```

Comment:
Sorry for the late comment, but what do you think about changing this order? The `killSwitchHandler` only checks a config boolean, so it is a cheap operation. If we first run the `concurrencyLimitHandler` then you could fill up the concurrency queue with requests to an endpoint that might not even be enabled.